internal/match/selector.go

```diff
@@ -0,0 +1,89 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package match
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// LabelSelector returns true if the selector set matches the values in the labels.
+func LabelSelector(selector metav1.LabelSelector, matchLabels map[string]string) bool {
```

Comment:
I think you should be able to use https://pkg.go.dev/k8s.io/apimachinery@v0.20.4/pkg/apis/meta/v1#LabelSelectorAsSelector and then https://pkg.go.dev/k8s.io/apimachinery/pkg/labels here to get the selector logic for free.

Reply:
oh nice! I was thinking it was already implemented but couldn't find it. Let me see how that would work.

Reply:
we used it in Velero, let me look for a link

Reply:
https://github.com/vmware-tanzu/velero/blob/0246a32ad0897053fbc444d86a0bec65762bc714/pkg/restore/restore.go#L160-L169
https://github.com/vmware-tanzu/velero/blob/0246a32ad0897053fbc444d86a0bec65762bc714/pkg/restore/restore.go#L769

Reply:
also, sorry i'm late here with the review :/

Reply:
nah no worries! Felt weird re-writing all that when it should have just been there (I should have asked).